ai2thor/util/visualize_3D_bbox.py

```diff
@@ -0,0 +1,248 @@
+                                pos_y = np.array(pos_x)
+                                pos_y[1] = pos_y[1] - (size[1] / 2) + (j * size[1])
+                                for k in range(2):
+                                    pos_z = np.array(pos_y)
+                                    pos_z[2] = pos_z[2] - (size[2] / 2) + (k * size[2])
+                                    corner_points.append(pos_z)
+
+                    # project vertices to 2D image coordinate
+                    for point in corner_points:
+                        new_point = project_to_agent_coordinate(point, agent_pos, rotation_matrix)
+                        if new_point[2] > 0:
+                            valid.append(True)
+                        else:
+                            valid.append(False)
+                        new_point_2d = project_to_2d(new_point, half_fov, w, h)
+                        vertices.append(new_point_2d)
+
+                    # get the 3D bbox 12 lines
+                    lines = [[vertices[0], vertices[1]],
+                             [vertices[2], vertices[3]],
+                             [vertices[0], vertices[2]],
+                             [vertices[1], vertices[3]],
+                             [vertices[4], vertices[5]],
+                             [vertices[6], vertices[7]],
+                             [vertices[4], vertices[6]],
+                             [vertices[5], vertices[7]],
+                             [vertices[2], vertices[6]],
+                             [vertices[3], vertices[7]],
+                             [vertices[1], vertices[5]],
+                             [vertices[0], vertices[4]]]
+                    valid_lines = [valid[0] * valid[1],
+                                   valid[2] * valid[3],
+                                   valid[0] * valid[2],
+                                   valid[1] * valid[3],
+                                   valid[4] * valid[5],
+                                   valid[6] * valid[7],
+                                   valid[4] * valid[6],
+                                   valid[5] * valid[7],
+                                   valid[2] * valid[6],
+                                   valid[3] * valid[7],
+                                   valid[1] * valid[5],
+                                   valid[0] * valid[4]]
+
+                lines = np.array(lines)
+                lines = np.reshape(lines, (-1, 4))
+                valid_lines = np.array(valid_lines)
+                valid_lines = np.reshape(valid_lines, (-1, 1))
+
+                # draw the 3D bbox 12 lines in the 2D RGB image
+                for iii, line in enumerate(lines):
+                    if valid_lines[iii]:
+                        draw.line((line[0], line[1], line[2], line[3]), fill=color, width=2)
+
+    # store the result back to the event
+    bbox_frame = np.array(img)
+    event.bbox_3d_frame = bbox_frame
+    return event
```

Comment:
Why is the event returned here? As well, modifying the event is not something a user would expect.  I think it would make more sense to return the bbox_frame.

Reply:
Another option, might be to move this code to a property named bbox_3d_frame off of the Event class in server.py.  Since the only dependency it has is the event it might make sense to move it there.

Reply:
Having a property named `bbox_3d_frame` in `server.py` sounds good to me!
I guess we can still keep the source script in `util/visualize_3D_bbox.py` and make a function call in `server.py`.
It might make more sense if we have more visualization tools in the future?